cargo-process.el

```diff
@@ -506,6 +537,61 @@ Requires Cargo clippy to be installed."
   (interactive)
   (cargo-process--start "Clippy" cargo-process--command-clippy))
 
+;;;###autoload
+(defun cargo-process-add (crate)
+  "Run the Cargo add command.
+With the prefix argument, modify the command's invocation.
+CRATES is the name of the crate to add.
+Cargo: This command allows you to add a dependency to a Cargo.toml manifest file."
+  (interactive (list
+				(completing-read "Crate(s) to add: " cargo-process-favorite-crates)))
+  (cargo-process--start "Add" (concat cargo-process--command-add
+									  " "
+									  crate)))
+
+;;;###autoload
+(defun cargo-process-rm (crate)
+  "Run the Cargo rm command.
+With the prefix argument, modify the command's invocation.
+CRATE is the name of the crate to remove.
+Cargo: Remove a dependency from a Cargo.toml manifest file."
+  (interactive
+   (list
+	(let ((deplist (cargo-process--get-dependencies
+					(concat
+					 (cargo-process--project-root) "Cargo.toml"))))
+	  (if deplist
+		  (completing-read "Crate to remove: "
+						   deplist nil t)
+		(progn
+		  (message "No crates used in current project.")
+		  nil)))))
+  (if crate (cargo-process--start "Remove" (concat cargo-process--command-rm
+												   " "
+												   crate))))
+;;;###autoload
+(defun cargo-process-upgrade (&optional all crates)
+  "Run the Cargo update command.
+With the prefix argument, modify the command's invocation.
+If ALL is t then update all crates, otherwise specify CRATES.
+Cargo: Upgrade dependencies as specified in the local manifest file"
+  (interactive)
+  (let ((deplist (cargo-process--get-dependencies
+				  (concat
+				   (cargo-process--project-root) "Cargo.toml"))))
+	(if deplist
+		(let ((all (when (or all
```

Comment:
the `crates` variable makes use of `all`, however you will need to define let as `let*` in order for it to be accessible to `crates`. Using only `let` will cause `all` to always default to the function argument, and never change (which will be nil almost always)